framework/src/utils/MooseUtils.C

```diff
@@ -578,6 +578,40 @@ getDefaultExecFlagEnum()
                               EXEC_CUSTOM);
   return exec_enum;
 }
+
+int
+stringToInteger(const std::string & input, bool throw_on_failure)
```

Comment:
Well this works, but it seems you took the hard way to get it done. I'll make a suggestion and see what you think.